Release 1.4.230 (#173)
mumble-voip/mumble-www: hugo/content/blog/2022-01-16-mumble-1-4-230.md

improvements.

You can download the new version from our [**Downloads page**]({{< relref "/downloads" >}}) or the
[**GitHub release page**](https://github.com/mumble-voip/mumble/releases/tag/v1.4.230) or from within your Windows client or software package
management system.

<!--more-->

# Mumble 1.4.230

It has been over two years since we released Mumble 1.3.0. Since then many new features and improvements have found their way into the software. And
not less importantly: many fixes for issues that have been encountered with Mumble. Talking about each and every change here would definitely exceed
the size of any reasonable release announcement. Therefore, we will restrict ourselves to highlighting what we think are the most prominent and/or
interesting changes and we invite everyone to simply try out the new version and explore what has changed yourself. For the interested reader, we of
course also include a full changelog at the bottom of this page as well.

If you are wondering, where versions 1.4.0 to 1.4.229 went, please have a look at a recent

> [blog post]({{< relref "/blog/2021-12-07-new-versioning-scheme/index.md" >}}) of us. In their we go into the details of our _new versioning scheme_ that

Reviewer: ```suggestion
[blog post]({{< relref "/blog/2021-12-07-new-versioning-scheme/index.md" >}}) of us. In there we go into the details of our _new versioning scheme_ that
```

> If you are wondering, where versions 1.4.0 to 1.4.229 went, please have a look at a recent
> [blog post]({{< relref "/blog/2021-12-07-new-versioning-scheme/index.md" >}}) of us. In their we go into the details of our _new versioning scheme_ that

Reviewer: ```suggestion
If you are wondering, why this first 1.4.x release is not versions 1.4.0, but 1.4.230, please see
[this recent blog post describing our _new versioning scheme_]({{< relref "/blog/2021-12-07-new-versioning-scheme/index.md" >}}) that
```

explains the absence of the mentioned version numbers.

## Feature spotlight

As there are many, many changes with this new version, here is a selection of things that we think are most prominent. If you're interested in all

> changes, have a look at the full [changelog](#changelog).

Reviewer: ```suggestion
changes, have a look at the [full changelog](#changelog).
```

*I was thinking of suggesting "further down" or similar, but decided against it.*

### Plugin framework

Mumble has supported something it called "plugins" for a very long time. However, these plugins were not what one would typically expect when hearing
about them for the first time. Instead of being general-purpose, (potentially) multi-function plugins, they were only ever dedicated for providing
positional data from games (generally by reading the game's in-memory representation). In addition to that, these plugins couldn't really be installed
and updated independently from the main application (for technical reasons) and therefore, plugins were mostly a static part of the Mumble client.

With Mumble 1.4 a new, general purpose plugin framework has been introduced. With this framework, the range of possibilities offered to plugin
developers has been vastly increased and most importantly plugins are no longer restricted to positional data delivery (though that is of course still
possible) and they can now be installed and updated at any time, without causing incompatibilities with the main Mumble client.

If you want to find out more about this new plugin framework (including how to write your first plugin), check out [the

> documentation](https://github.com/mumble-voip/mumble/tree/master/docs/dev/plugins).

Reviewer: I would label the link `the plugin documentation` for a more specific and independently specific hyperlink - which should also improve SEO.
```suggestion
plugin documentation](https://github.com/mumble-voip/mumble/tree/master/docs/dev/plugins).
```

per-user basis.

Concretely this new feature allows a user to "listen to" a channel. In that case all audio that is heard by people in this particular channel (be it
by direct communication, shouts or via linked channels) is also heard by the listening user. Conceptually you could picture "listening to" a channel
with placing a microphone in a room and then listening to what is recorded.

This feature is one-way only, meaning that in order to speak to people in the listened channel, you'll have to either join the channel or shout to it.

It should be noted that you will not be secretly listened to with this new feature. Listening to a channel creates a new entry in the channel's
user-list (referred to as an "ear" or a "listener") and is thereby just as visible as a user actually being in that channel. There is also a dedicated
ACL and several new server config options that can restrict this new feature.

Older clients (< 1.4) will not indicate listeners in the UI though! However, they are informed if this feature is enabled on a server when they join
it (via chat).

> Note: This feature requires server-side support and therefore is only available on servers at least running Mumble 1.4.

Reviewer: ```suggestion
Note: This feature requires server-side support and therefore is only available on servers running at least Mumble 1.4.
```

Note that this is also a feature that requires server-side support and thus only takes effect on servers that are at least running Mumble version
1.4.

![Access restricted channels](/blog/media/AccessRestrictedChannels.png)

### Markdown

For those of you who don't know what markdown is, we recommend a quick read of the respective [Wikipedia
article](https://en.wikipedia.org/wiki/Markdown). In simple words it allows formatting of text by simply typing a few extra characters in it.

Mumble now supports markdown for text messages written in the normal chat bar. It is not supported for cases in which you use the dedicated message
editor (e.g. for private messages). Furthermore only a subset of markdown is supported: sections, inline-code, code-blocks, bold, italic and
strike-through text, links and quotes. _Not_ supported are lists and tables.

This feature is probably best explained by an example. Try pasting the following text into the chat bar and see for yourself:

````
# Markdown
## Code
We can have `inline` code and
```
code
    blocks
```

## Emphasis
**bold** and *italic* (__bold__ and _italic_ also works) text is supported. ~~Scratch that~~

## Links

> We can have [links](google.de) and www.google.de

Reviewer: We could use a neutral website; our own website.
```suggestion
We can have [labeled links](https://www.mumble.info) and www.mumble.info
```

Author: But then the screenshot doesn't match anymore - and in the end I don't think it really matters, does it? I just used the same example that I used in my original PR :shrug:

Note however that this feature is currently restricted to playback. The official Mumble client will still continue sending audio as mono. This feature
is primarily intended to be made use of by (music) bots.

If you want to try this feature out yourself, you can either write a bot using [pymumble](https://github.com/azlux/pymumble) version 1.2 or later or
you can simply use a recent version of [botamusqiue](https://github.com/azlux/botamusique).

### Nicknames

For those of you who are annoyed by people using excessively long or complicated names or people changing their nickname frequently, Mumble 1.4
introduces the possibility to set a nickname for users.

Depending on your configuration, a nickname will either be displayed next to a user's name or replace it entirely.

The nickname will persist across restarts of client and server. In fact it'll last until you either remove it again or the person the nickname was
assigned to changes their client-certificate (Mumble's way of keeping track of a user's identity).

Note that nicknames are _not_ tied to a particular server. That means that if you assign a nickname to a user on server A and then you meet that

> person again on server B, they will still have that very same nickname set on server B.

Reviewer: We could mention that this is like it was before for users added to the friend list. For example:

This behavior is the same as the friend list feature. After adding a user as a friend on server A, you could identify them on server B by their known nickname even when they use a different nickname on that server B.
